lib/devise/strategies/http_header_authenticatable.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'devise/strategies/authenticatable'
+module Devise
+  module Strategies
+    class HttpHeaderAuthenticatable < Authenticatable
+      def authenticate!
+        return pass if request.session[:user_name].present?
+
+        http_user = remote_user(request.headers)
+        return fail! if http_user.blank?
+
+        user = User.find_or_create_by(email: http_user)
+        success!(user)
+      end
+
+      def valid?
```

Comment:
You should be able to make this method a one liner: `remote_user(request.headers).present?`

Reply:
Good catch.